sklearn/gaussian_process/gpr.py

```diff
@@ -0,0 +1,148 @@
+"""Gaussian processes regression. """
+
+# Authors: Jan Hendrik Metzen <jhm@informatik.uni-bremen.de>
+#
+# License: BSD 3 clause
+
+import numpy as np
+from scipy.linalg import cholesky, cho_solve, solve
+from scipy.optimize import fmin_l_bfgs_b
+
+from sklearn.base import BaseEstimator
+
+
+class GaussianProcessRegression(BaseEstimator):
+    """ Gaussian process regression (GPR).
+
+    The implementation is based on Algorithm 2.1 of ``Gaussian Processes
+    for Machine Learning'' (GPML) by Rasmussen and Williams.
+
+    In addition to standard sklearn estimators, GaussianProcessRegression
+       * allows prediction without prior fitting (based on the GP prior)
+       * provides an additional method sample(X), which evaluates samples drawn
+         from the GPR (prior or posterior) at given inputs
+       * exposes a method log_marginal_likelihood(theta), which can be used
+         externally for other ways of selecting hyperparamters, e.g., via
+         Markov chain Monte Carlo.
+
+    Parameters
+    ----------
+    kernel : Kernel object
+        The kernel specifying the covariance function of the GP.
+
+    y_err : float, optional (default: 1e-10)
+        Value added to the diagonal of the kernel matrix during fitting.
+        Larger values correspond to increased noise level in the observations
+        and reduce potential numerical issue during fitting.
+
+    Attributes
+    ----------
+    X_fit_:
+
+    y_fit_:
+
+    theta_:
+
+    L_:
+
+    alpha_:
+    """
+
+    def __init__(self, kernel, y_err=1e-10):
+        self.kernel = kernel
+        self.y_err = y_err
```

Comment:
I guess we are in a different setting here, and thus nomenclature is different, but I am identifying `y_err` as a regularization parameter (coming from the point of view of somebody who does ridge regression). So I am for having at least an explanation referring to the link between gpr and krr or a renaming that is acceptable for both domains.